### PR TITLE
Read Scalyr API key from file

### PR DESCRIPTION
**Pull request by anton-ryzhov**
Read Scalyr API key from file to update it without reloading
Use pip for building an image

### CODE REVIEW

**Comment by anton-ryzhov** [comment]
:+1:

File: tests/test_scalyr.py [hunk 1]
```diff
@@ -207,32 +217,32 @@ def test_initialization_failure(monkeypatch, env, exists):
...
+    monkeypatch.setattr('os.path.isdir', isdir)
 
     exists = MagicMock()
-    exists.side_effect = (True, True, True, False, False, True)
+    exists.side_effect = [True, False, False, True]
     monkeypatch.setattr('os.path.exists', exists)
 
     makedirs, symlink, listdir = patch_os(monkeypatch)
 
     current_targets = MagicMock()
```
Comment (pheanex) [inline tests/test_scalyr.py]:
```suggestion
    current_targets = MagicMock(return_value=set())
```

File: tests/test_scalyr.py [hunk 2]
```diff
@@ -254,42 +264,43 @@ def test_add_log_target(monkeypatch, env, fx_scalyr):
...
+def test_add_log_target_no_src(monkeypatch, scalyr_env, fx_scalyr):
     patch_os(monkeypatch)
-    patch_env(monkeypatch, env)
 
     target = fx_scalyr['target']
 
+    isdir = MagicMock(side_effect=[True, True])
+    monkeypatch.setattr('os.path.isdir', isdir)
+
     exists = MagicMock()
```
Comment (pheanex) [inline tests/test_scalyr.py]:
```suggestion
    exists = MagicMock(side_effect=[False])
```

Comment (pheanex) [reply tests/test_scalyr.py]: Including all of those occurrences below

File: tests/test_scalyr.py [hunk 3]
```diff
@@ -207,32 +217,32 @@ def test_initialization_failure(monkeypatch, env, exists):
...
-
+def test_add_log_target(monkeypatch, scalyr_env, fx_scalyr):
     target = fx_scalyr['target']
     kwargs = fx_scalyr['kwargs']
 
     assert_fx_sanity(kwargs)
 
     # adjust kwargs
-    kwargs['monitor_journald'] = {} if not env.get('WATCHER_SCALYR_JOURNALD') else SCALYR_MONITOR_JOURNALD
+    kwargs['monitor_journald'] = {} if not os.environ.get('WATCHER_SCALYR_JOURNALD') else SCALYR_MONITOR_JOURNALD
```
[the remark this answers is not in the document]
Comment (anton-ryzhov) [reply tests/test_scalyr.py]: Sure! I just did search&replace and missed inverted logic here

File: tests/test_scalyr.py (hunk 2 above)
Comment (anton-ryzhov) [reply tests/test_scalyr.py]: That's a lot:
```
$ ag -c 'MagicMock\(\)'
tests/test_main.py:27
tests/test_appdynamics.py:4
tests/test_base.py:1
tests/test_kube.py:8
tests/test_scalyr.py:19
```

And I don't like this patching at all. It's better to replace it with temp files. But do we want to rewrite it now?

Comment (anton-ryzhov) [reply tests/test_scalyr.py]: Oh, you've just inlined already defined `side_effect`s. That's an easy change

**Comment by pheanex** [comment]
:+1:

**Comment by anton-ryzhov** [comment]
:+1:

**Pull request merged**